libs/content/workspace-ffi/src/apple/ios/api.rs

```diff
@@ -16,12 +17,21 @@ use workspace_rs::tab::TabContent;
 use super::super::response::*;
 use super::response::*;
 use crate::apple::keyboard::UIKeys;
-use crate::WgpuWorkspace;
+use crate::{ExtendedInput, WgpuWorkspace};
 
 #[no_mangle]
 pub extern "C" fn ios_frame(obj: *mut c_void) -> IOSResponse {
     let obj = unsafe { &mut *(obj as *mut WgpuWorkspace) };
-    obj.frame().into()
+
+    let frame = obj.context.frame_nr();
+    let span = span!(target: "svg_editor", Level::TRACE, "request ios frame", frame);
+    let _ = span.enter();
+
+    let res = obj.frame().into();
+
+    obj.context.push_span(span);
```

Comment:
I think you do not need to hang onto spans, I'm fairly certain all the behavior you're describing in this PR can happen automatically